sotodlib/io/load_smurf.py

```diff
@@ -96,6 +96,35 @@ def get_sample_timestamps(frame):
                           for t in frame['data'].times()])
         return times, TimingParadigm.G3Timestream
 
+def _file_has_end_frames(filename):
+    ended = False
+    
+    ## Search through file looking for stream closeout
+    reader = so3g.G3IndexedReader(filename)
+    logger.debug(f"Reading {filename} to look for observation end.")
+    while True:
```

Comment:
You know significantly more about the file structure than me, so you probably know this loop will always terminate, but just making sure these conditionals will catch every possible outcome.

Reply:
yeah, it will terminate at the end of the file no matter what